src/wallet/test/fuzz/scriptpubkeyman.cpp

```diff
@@ -200,5 +207,88 @@ FUZZ_TARGET(scriptpubkeyman, .init = initialize_spkm)
+    }
+
+    std::vector<CKey> keys;
+    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 30) {
+        const auto key{ConsumePrivateKey(fuzzed_data_provider)};
+        if (!key.IsValid()) return;
+        keys.push_back(key);
+    }
+
+    if (keys.empty()) return;
+
+    auto& legacy_data{*wallet.GetOrCreateLegacyDataSPKM()};
+
+    int load_key_count{0};
+    int script_count{0};
+
+    bool good_data{true};
+    LIMITED_WHILE(good_data && fuzzed_data_provider.ConsumeBool(), 30) {
+        CallOneOf(
+            fuzzed_data_provider,
+            [&] {
+                const auto key_index{fuzzed_data_provider.ConsumeIntegralInRange<int>(0, keys.size() - 1)};
+                if (legacy_data.HaveKey(keys[key_index].GetPubKey().GetID())) return;
+                if (legacy_data.LoadKey(keys[key_index], keys[key_index].GetPubKey())) {
+                    load_key_count++;
+                    if (fuzzed_data_provider.ConsumeBool()) {
+                        CHDChain hd_chain;
+                        hd_chain.nVersion = CHDChain::VERSION_HD_CHAIN_SPLIT;
+                        hd_chain.seed_id = keys[key_index].GetPubKey().GetID();
+                        legacy_data.LoadHDChain(hd_chain);
+                    }
+                }
+            },
+            [&] {
+                std::optional<CKeyMetadata> key_metadata{ConsumeDeserializable<CKeyMetadata>(fuzzed_data_provider)};
+                if (!key_metadata) {
+                    good_data = false;
+                    return;
+                }
+                const auto key_index{fuzzed_data_provider.ConsumeIntegralInRange<int>(0, keys.size() - 1)};
+                auto key_id{keys[key_index].GetPubKey().GetID()};
+                legacy_data.LoadKeyMetadata(key_id, *key_metadata);
+            },
+            [&] {
+                CScript script{ConsumeScript(fuzzed_data_provider)};
+                const auto key_index{fuzzed_data_provider.ConsumeIntegralInRange<int>(0, keys.size() - 1)};
+                LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 2) {
+                    CallOneOf(
+                        fuzzed_data_provider,
+                        [&] {
+                            script = GetScriptForDestination(PKHash(keys[key_index].GetPubKey()));
+                        },
+                        [&] {
+                            script = GetScriptForDestination(WitnessV0KeyHash(keys[key_index].GetPubKey()));
+                        },
+                        [&] {
+                            script = GetScriptForDestination(WitnessV0ScriptHash(script));
+                        }
+                    );
+                }
+                if (legacy_data.AddCScript(script)) script_count++;
+            }
+        );
+    }
+
+    std::optional<MigrationData> res{legacy_data.MigrateToDescriptor()};
+    if (res.has_value()) {
+        assert(static_cast<int>(res->desc_spkms.size()) >= load_key_count);
```

Comment:
If watchonly script count can be tracked, then this check should be able to be an exact check as `desc_spkms` should be `load_key_count` plus the number of spendable scripts.

Reply:
Done.